src/index.js

```diff
@@ -39,7 +39,14 @@ if (
 }
 
 /* Warning if there are several instances of styled-components */
-if (process.env.NODE_ENV !== 'production' && typeof window !== 'undefined') {
+if (
+  process.env.NODE_ENV !== 'production' &&
+  process.env.NODE_ENV !== 'test' &&
+  typeof window !== 'undefined' &&
+  typeof navigator !== 'undefined' &&
+  navigator.userAgent.indexOf('Node.js') !== -1 &&
+  navigator.userAgent.indexOf('jsdom') !== -1
+) {
   window['__styled-components-init__'] =
```

Comment:
While you're at it. Shall we simplify this? There's no need to count them up

Reply:
Just want to note that I added the counting because there is a chance of getting multiple warnings otherwise

Reply:
That's a good point! didn't think about that :+1: then I'd only extract the `'__styled-components-init__'` string into a variable and call it a day 😸